File: CS-Examples/02_Data/CopyOnlyFormulaValue.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CopyOnlyFormulaValue
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook
            Workbook workbook = new Workbook();

            // Load an existing workbook from a file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\CopyOnlyFormulaValue.xlsx");

            // Get the first worksheet in the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Set the copy option to only copy the formula values
            CopyRangeOptions copyOptions = CopyRangeOptions.OnlyCopyFormulaValue;

            // Copy a range of cells from A2:C2 to A5:C5 using the specified copy options
            sheet.Copy(sheet.Range["A2:C2"], sheet.Range["A5:C5"], copyOptions);

            // Save the modified workbook to a new file named "result.xlsx" in Excel 2010 format
            workbook.SaveToFile("result.xlsx", ExcelVersion.Version2010);

            // Dispose of the workbook object to free up resources
            workbook.Dispose();

            //View the document
            FileViewer("result.xlsx");
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/02_Data/CopyOnlyFormulaValue.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace CopyOnlyFormulaValue
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            '  Create a new workbook object.
            Dim workbook As New Workbook()

            ' Load an Excel document named "CopyOnlyFormulaValue.xlsx" from the specified file path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\CopyOnlyFormulaValue.xlsx")

            ' Get the first worksheet from the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Specify the copy option as OnlyCopyFormulaValue, which means only the formula values will be copied.
            Dim copyOptions As CopyRangeOptions = CopyRangeOptions.OnlyCopyFormulaValue

            ' Copy the range from A2 to C2 and paste it to A5 to C5 in the same worksheet, using the specified copy options.
            sheet.Copy(sheet.Range("A2:C2"), sheet.Range("A5:C5"), copyOptions)

            ' Save the modified workbook to a file named "result.xlsx" using Excel 2010 format.
            workbook.SaveToFile("result.xlsx", ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer("result.xlsx")
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
